src/github/client.rs

```diff
@@ -116,86 +106,45 @@ impl Client {
         params.insert("since", format!("{:?}", start));
         params.insert("per_page", format!("{}", PER_PAGE));
 
-        self.get_models(&url, &params)
+        self.get_models(&url, Some(&params))
     }
 
     fn get_models<M: Deserialize>(&self,
                                   start_url: &str,
-                                  params: &ParameterMap)
+                                  params: Option<&ParameterMap>)
                                   -> DashResult<Vec<M>> {
-        let mut res = try!(self.get(start_url, Some(&params)));
-
-        // let's try deserializing!
-        let mut buf = String::new();
-        try!(res.read_to_string(&mut buf));
-
-        let mut models = match serde_json::from_str::<Vec<M>>(&buf) {
-            Ok(m) => m,
-            Err(why) => {
-                error!("Unable to parse from JSON ({:?}): {}", &why, buf);
-                return Err(why.into());
-            }
-        };
 
-        let mut next_url = Self::next_page(&res.headers);
-        while next_url.is_some() {
-            // TODO figure out a better rate limit
+        let mut res = try!(self.get(start_url, params));
```

Comment:
Doesn't need changing but small style nit: I think most new code in the repo has begun moving to question mark syntax now that it's stable.